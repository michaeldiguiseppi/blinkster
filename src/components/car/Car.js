import React, { Component } from 'react';
import moment from 'moment';
import PropTypes from 'prop-types';

export class Car extends Component {
    constructor(props) {
        super(props);

        this.handleClick = this.handleClick.bind(this);
    }

    handleClick() {
        let { car } = this.props;
        this.context.router.history.push(`/cars/${car.year}/${car.make}/${car.model}`);
    }

   render() {
        let { car } = this.props;
        return (
            <tr onClick={ this.handleClick }>
                <th scope="row">{ car.year }</th>
                <td>{ car.make }</td>
                <td>{ car.model }</td>
                <td>{ car.mileage }</td>
                <td className="veh-drivetrain">{ car.drivetrain ? car.drivetrain : "N/A" }</td>
                <td>{ `${moment(car.created_at).format("MMMM Do, YYYY")}` }</td>
            </tr>
        )
    }

    static contextTypes = {
        router: PropTypes.shape({
          history: PropTypes.shape({
            push: PropTypes.func.isRequired,
          }),
          staticContext: PropTypes.object
        })
      };
}

export default Car;