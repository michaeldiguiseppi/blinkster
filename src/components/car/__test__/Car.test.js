import React from 'react';
import { Car } from '../Car';
import renderer from 'react-test-renderer';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('<Car /> component', () => {
    let carComponent;
    const defaultProps = {
        car: {
                "year" : 2013,
                "make" : "Kia",
                "model" : "Optima",
                "mileage" : 24235,
                "drivetrain" : "FWD",
                "bodytype" : "sedan",
                "image_url" : "http://www.optimaforums.com/forum/attachments/new-member-introductions/11137d1347548855-new-2013-kia-optima-sx-l-titanium-photo.jpg",
                "created_at" : "2016-10-14T20:13:22.586Z"
        },
        index: 0
    }

    beforeEach(() => {
        carComponent = mount(<table><tbody><Car {...defaultProps} /></tbody></table>);
    });

    afterEach(() => {
        carComponent.unmount();
    });

    it('should render without breaking', () => {
        expect(carComponent.exists()).toBe(true);
    });

    it('should match the snapshot', () => {
        const tree = renderer
            .create(<table><tbody><Car {...defaultProps} /></tbody></table>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render the drivetrain if it exists', () => {
        expect(carComponent.find('.veh-drivetrain').exists()).toBe(true);
        expect(carComponent.find('.veh-drivetrain').text()).toEqual("FWD");
    });

    it('should render "N/A" if no drivetrain exists', () => {
        let props = {
            car: {
                "year" : 2013,
                "make" : "Kia",
                "model" : "Optima",
                "mileage" : 24235,
                "bodytype" : "sedan",
                "image_url" : "http://www.optimaforums.com/forum/attachments/new-member-introductions/11137d1347548855-new-2013-kia-optima-sx-l-titanium-photo.jpg",
                "created_at" : "2016-10-14T20:13:22.586Z"
        },
        index: 0
        }
        carComponent.unmount();
        carComponent = mount(<table><tbody><Car {...props} /></tbody></table>);
        expect(carComponent.find('.veh-drivetrain').text()).toEqual("N/A");
    });
});