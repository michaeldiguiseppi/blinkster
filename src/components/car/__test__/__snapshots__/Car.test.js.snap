// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Car /> component should match the snapshot 1`] = `
<table>
  <tbody>
    <tr
      onClick={[Function]}
    >
      <th
        scope="row"
      >
        2013
      </th>
      <td>
        Kia
      </td>
      <td>
        Optima
      </td>
      <td>
        24235
      </td>
      <td
        className="veh-drivetrain"
      >
        FWD
      </td>
      <td>
        October 14th, 2016
      </td>
    </tr>
  </tbody>
</table>
`;
